plenum/test/checkpoints/test_checkpoints_removal_in_view_change.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+import sys
+
+from plenum.common.constants import CHECKPOINT, COMMIT
+from plenum.test.delayers import cDelay, chk_delay
+from plenum.test.helper import sdk_send_random_requests, \
+    sdk_get_and_check_replies, sdk_send_random_and_check
+from plenum.test.node_catchup.helper import ensure_all_nodes_have_same_data
+from plenum.test.test_node import ensureElectionsDone
+from stp_core.loop.eventually import eventually
+
+CHK_FREQ = 2
+
+
+def test_checkpoints_removed_in_view_change(chkFreqPatched,
+                                            txnPoolNodeSet,
+                                            looper,
+                                            sdk_pool_handle,
+                                            sdk_wallet_client):
+    '''
+    Check that checkpoint finalize in view change before catchup doesn't clean
+    necessary data from requests and 3pc queues.
+    '''
+    slow_nodes = txnPoolNodeSet[:3]
+    fast_nodes = txnPoolNodeSet[3:]
+    # delay checkpoints processing for slow_nodes
+    delay_msg(slow_nodes, chk_delay)
+    # send txns for finalizing current checkpoint
+    sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle,
+                              sdk_wallet_client, CHK_FREQ)
+    ensure_all_nodes_have_same_data(looper, nodes=txnPoolNodeSet)
+    # delay commits processing for slow_nodes
+    delay_msg(slow_nodes, cDelay)
+
+    requests = sdk_send_random_requests(looper, sdk_pool_handle,
+                                         sdk_wallet_client, 1)
+    # check that slow nodes have prepared certificate with new txn
+    looper.run(eventually(last_prepared_certificate,
+                          slow_nodes,
+                          (0, CHK_FREQ + 1)))
+    # check that fast_nodes ordered new txn
+    looper.run(eventually(last_ordered_check,
+                          fast_nodes,
+                          (0, CHK_FREQ + 1)))
+    # check that fast_nodes finalized first checkpoint
+    looper.run(eventually(check_checkpoint_finish,
```

Comment:
Please check that checkpoint is not finalized for slow_nodes

Reply:
Done